inginious/frontend/pages/lti.py

```diff
@@ -112,9 +112,26 @@ def GET(self):
         except:
             return self.template_helper.get_renderer().lti_bind(False, "", None, "Invalid LTI data")
 
-        user_profile = self.database.users.find_one({"ltibindings." + data["task"][0] + "." + data["consumer_key"]: data["username"]})
+        if course.lti_auto_bind():
+            lti_email = data['email']
+            lti_name = data['realname']
+            lti_id = lti_email[0:lti_email.find('@')]
+            user_profile = self.database.users.find_one({"username": lti_id})
```

Comment:
I'm afraid this will lead to a severe security issue enabling account hijacking. This may seem harmless in a closed environment such as a university Moodle instance where all students have the same email domain, but in case an LTI course is used in a public environment (such as our edX Louv1.x on inginious.org), ``somebody@somedomain`` may be bound to the ``somebody@anotherdomain`` account, or, actually, simply an alreay existing account with corresponding user id.

A more secure and acceptable auto-binding feature would be to silently accept the binding in place of the user **once logged in**.

Reply:
Of course this is right and this feature can lead to serve security issues, but as you can see, this feature has to be manually activated in the ```course.yaml```.
Solutions regarding to the other points you made:
- whitelisting only one domain for this feature
- implement mapping of domains to some pre-/postfixes (```alice@gmail.com``` -> ```alice_gmail```, ```alice@company.com``` -> ```alice_company```). I know, admins will have to keep track over their mappings to ensure the uniqueness of the name.

What would you think, would be the best solution? If necessary, I will implement one of these features or something else, if you have better ideas.